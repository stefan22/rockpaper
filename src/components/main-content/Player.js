import React from 'react';
import PropTypes from 'prop-types';
import Card from '@material-ui/core/Card';
import {withStyles} from '@material-ui/core/styles';
import Avatar from '@material-ui/core/Avatar';
import Grid from '@material-ui/core/Grid';
import '../../scss/markets.scss';

const styles = {
  avatar: {
    margin: '10px',
  },
  card: {
    minWidth: '275px',
    padding: '10px',
    cursor: 'pointer',
  },
  span: {
    color: '#2196F3',
  },
  bigAvatar: {
    margin: '10px',
    width: '60px',
    height: '60px',
  },
};

const Player = (props) => {
  let {classes,name,imgsrc,playerNumber} = props;
  return (
    <section className='col-4 players-avatar'>
      <Card className={classes.card}
        onClick={() => props.onPlayerChoice(props.playerNumber)}
      >
        <Grid container justify="center" alignItems="center">
          <Avatar alt={name} src={imgsrc} className={classes.bigAvatar} />
          <span className={classes.span}>{props.name}</span>

        </Grid>
        <span>as player {playerNumber}</span>
      </Card>
    </section>
  );
};

Player.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(Player);
