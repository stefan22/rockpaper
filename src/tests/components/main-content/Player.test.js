import React from 'react';
import {shallow} from 'enzyme';
import Player from '../../../components/main-content/Player';

describe('Player component', () => {
  it('should render player correctly', () => {
    const wrapper = shallow(<Player />);
    expect(wrapper).toMatchSnapshot();
  });




});
