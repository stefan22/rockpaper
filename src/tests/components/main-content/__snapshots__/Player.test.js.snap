// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player component should render player correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(Player) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "classes": Object {
        "avatar": "Player-avatar-1",
        "bigAvatar": "Player-bigAvatar-4",
        "card": "Player-card-2",
        "span": "Player-span-3",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "classes": Object {
          "avatar": "Player-avatar-1",
          "bigAvatar": "Player-bigAvatar-4",
          "card": "Player-card-2",
          "span": "Player-span-3",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
